src/components/PopupWithForm.js

```diff
@@ -1,31 +1,44 @@
-import Popup from '../components/Popup';
+import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
-    super({ popupSelector });
-    this._handleFormSubmit = handleFormSubmit;
+  constructor(popupSelector, handleSubmit) {
+    super(popupSelector);
+    this._handleSubmit = handleSubmit;
     this._popupForm = this._popupElement.querySelector('.modal__form');
   }
 
   _getInputValues() {
-    const inputs = this._popupForm.querySelectorAll('.modal__input');
-    const values = {};
-    inputs.forEach(input => {
-      values[input.name] = input.value;
+    const inputValues = {};
+    this._inputList = this._popupForm.querySelectorAll('.modal__input');
+    this._inputList.forEach((input) => {
+      if (input.value !== "") {
+        inputObject[input.name] = input.value;
+      }
     });
-    return values;
+    return inputValues
   }
 
-  setEventListeners() {
-    this._popupForm.addEventListener('submit', () => {
-      const inputValues = this._getInputValues();
-      this._handleFormSubmit(inputValues);
-    });
-    super.setEventListeners();
+  _handleSubmit(e) {
+    e.preventDefault();
+    this._handleSubmit(this._getInputValues());
   }
 
   close() {
-    this._popupForm.reset();
     super.close();
+    this._popupForm.reset();
+    this._popupForm.removeEventListener('submit', this._handleSubmit);
```

Comment:
There is no need to remove the listener here, since you add it only once in `setEventListeners` method